liquibase-core/src/main/java/liquibase/integration/servlet/GenericServletListener.java

```diff
@@ -0,0 +1,256 @@
+    private static final String LIQUIBASE_PARAMETER = "liquibase.parameter";
+    private static final String LIQUIBASE_SCHEMA_DEFAULT = "liquibase.schema.default";
+
+    private String changeLogFile;
+    private String dataSourceName;
+    private String contexts;
+    private String labels;
+    private String defaultSchema;
+    private String hostName;
+
+    public String getChangeLogFile() {
+        return changeLogFile;
+    }
+
+    public void setChangeLogFile(String changeLogFile) {
+        this.changeLogFile = changeLogFile;
+    }
+
+    public String getContexts() {
+        return contexts;
+    }
+
+    public void setContexts(String ctxt) {
+        contexts = ctxt;
+    }
+
+    public String getLabels() {
+        return labels;
+    }
+
+    public void setLabels(String labels) {
+        this.labels = labels;
+    }
+
+    public String getDataSource() {
+        return dataSourceName;
+    }
+
+    /**
+     * Sets the name of the data source.
+     */
+    public void setDataSource(String dataSource) {
+        this.dataSourceName = dataSource;
+    }
+
+    public String getDefaultSchema() {
+        return defaultSchema;
+    }
+
+    public void contextInitialized(GenericServletWrapper.ServletContext servletContext) {
+        final LiquibaseConfiguration liquibaseConfiguration = Scope.getCurrentScope().getSingleton(LiquibaseConfiguration.class);
+
+        try {
+            this.hostName = NetUtil.getLocalHostName();
+        } catch (Exception e) {
+            servletContext.log("Cannot find hostname: " + e.getMessage());
+            return;
+        }
+
+        InitialContext ic = null;
+        String failOnError = null;
+        final ServletConfigurationValueProvider servletConfigurationValueProvider = new ServletConfigurationValueProvider(servletContext, ic);
+        try {
+            ic = new InitialContext();
+
+            liquibaseConfiguration.registerProvider(servletConfigurationValueProvider);
+
+            failOnError = (String) liquibaseConfiguration.getCurrentConfiguredValue(null, null, LIQUIBASE_ONERROR_FAIL).getValue();
+            if (checkPreconditions(servletContext, ic)) {
+                executeUpdate(servletContext, ic);
+            }
+
+        } catch (Exception e) {
+            if (!"false".equals(failOnError)) {
+                throw new RuntimeException(e);
+            }
+        } finally {
+            if (ic != null) {
+                try {
+                    ic.close();
+                } catch (NamingException e) {
+                    // ignore
+                }
+            }
+            liquibaseConfiguration.removeProvider(servletConfigurationValueProvider);
+
+
+        }
+    }
+
+    /**
+     * Checks if the update is supposed to be executed. That depends on several conditions:
+     * <ol>
+     * <li>if liquibase.shouldRun is <code>false</code> the update will not be executed.</li>
```

Comment:
I see now how those parameters are used and also infer that these have no cognate in the CLI. :)

Reply:
Right, we didn't bother exposing them in the CLI so far since that is more explicitly requested. As we continue to clean up the integrations for consistency, the arguments may eventually show up in the CLI